docs/protocol.rst

```diff
@@ -0,0 +1,60 @@
+Tunneldigger protocol
+#####################
+
+The Tunneldigger protocol is based on L2TPv3.
+Tunneldigger implementing a custom control protocol,
+while user data is encapsulated as L2TPv3 specifies.
+This allows to use in kernel acceleration of the data path.
+
+The L2TPv3 is specified in `RFC3931 <https://tools.ietf.org/html/rfc3931>`_
+
+Control packets
+***************
+
+Tunneldigger is not following RFC3931 for control messages.
+Tunneldigger is only supporting the T bit in the first bit to
+mark a packet as control data. All other fields of RFC3931
+are ignore and have a different meaning..
+All fields are encoding as network byte order.
+
+.. packetdiag::
+
+    {
+        colwidth = 16
+        node_height = 48
+
+        0:      T
+        1-7:    0
+        8-23:   Magic 0x73A7
+        24-31:  Version
+        32-39:  Type
+        40-47:  Length
+        48-63:  Value
```

Comment:
The length of value depends on the type; why do you write `48-63` here?

Reply:
Because it's a packet diagraph. I would like to have a value in there. It's only an example.

Reply:
The problem is that it is not at all clear this is just an example. And that makes it very confusing IMO.